src/redux/books/books.js

```diff
@@ -1,63 +1,76 @@
-import { v4 as uuidv4 } from 'uuid';
+import { createAsyncThunk } from '@reduxjs/toolkit';
 
 // Action types
 const addBook = 'bookStore/books/addBook';
 const removeBook = 'bookStore/books/removeBook';
-const mockBookList = [
-  {
-    id: uuidv4(),
-    schoolOf: 'Action',
-    title: 'The Hunger Games',
-    author: 'Suzanne Collins',
-    percentComplete: '64',
-    currentChapter: 'Chapter 17',
-  },
-  {
-    id: uuidv4(),
-    schoolOf: 'Science Fiction',
-    title: 'Dune',
-    author: 'Frank Herbert',
-    percentComplete: '8',
-    currentChapter: 'Chapter 1',
-  },
-  {
-    id: uuidv4(),
-    schoolOf: 'Economy',
-    title: 'Capital in the Twenty-First Century',
-    author: 'Suzanne Collins',
-    percentComplete: '0',
-    currentChapter: 'Introduction',
-  },
-];
-
-const initialState = mockBookList;
-
-export const addBookAction = (title, author) => ({
-  type: addBook,
-  payload: {
-    id: uuidv4(),
-    schoolOf: 'Action',
-    title,
-    author,
-    percentComplete: 0,
-    currentChapter: 'Introduction',
-  },
-});
+const getBooks = 'bookStore/books/getBooks';
 
-export const removeBookAction = (id) => ({
-  type: removeBook,
-  id,
-});
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
+const apiId = 'JUWUUSwLtCmHou4Ag9hk';
+
+const initialState = {
+  books: [],
+  status: null,
+};
 
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
-    case addBook:
-      return [...state, action.payload];
-    case removeBook:
-      return state.filter((item) => item.id !== action.id);
+    case 'bookStore/books/getBooks/fulfilled':
+      return {
+        books: action.payload,
+        status: 'Fetch book list succeded',
+      };
+    case 'bookStore/books/addBook/fulfilled':
+      return {
+        books: [...state.books, action.payload],
+        status: 'Add book succeded',
+      };
+
+    case 'bookStore/books/removeBook/fulfilled':
+      return {
+        books: state.books.filter((item) => item.item_id !== action.payload),
+        status: 'Delete book succeded',
+      };
     default:
       return state;
   }
 };
 
+export const getBooksAction = createAsyncThunk(getBooks, async () => {
+  const response = await fetch(`${baseUrl}/apps/${apiId}/books`);
+  const data = await response.json();
+  const keys = Object.keys(data);
+  const arrayData = [];
+  keys.map((key) => arrayData.push({
+    item_id: key,
+    title: data[key][0].title,
+    author: data[key][0].author,
+    category: data[key][0].category,
+  }));
+  return arrayData || [];
+});
+
+export const addBookAction = createAsyncThunk(addBook, async (book) => {
+  await fetch(`${baseUrl}/apps/${apiId}/books`, {
```

Comment:
When I add a book it doesn't appear on the page until I refresh the page. This is because the new book is not simultaneously added to the local state, it is only added to the API.

If you look at the screenshots below, I logged your actions. You can see that the `addBook` action is fulfilled and yet the local state `books` is still the initial state.

![image](https://user-images.githubusercontent.com/38088818/187020508-10ceba16-d9aa-4c63-8a80-68f80bea152a.png)

Until I refresh the page and the `getBooks` action is fired to fetch books from the API, then the local state is updated.

![image](https://user-images.githubusercontent.com/38088818/187020678-6b0c9574-351f-4ed4-864b-6d209b66a662.png)

- Kindly make sure that the local state is updated with the new book even before I reload the page.